service/gcs/core/gcs/uvm.go

```diff
@@ -96,6 +100,80 @@ func (h *Host) CreateContainer(id string, settings *prot.VMHostedContainerSettin
 		return c, nil
 	}
 
+	isSandboxOrStandalone := false
+	if criType, ok := settings.OCISpecification.Annotations["io.kubernetes.cri.container-type"]; ok {
+		// If the annotation is present it must be "sandbox"
+		isSandboxOrStandalone = criType == "sandbox"
+	} else {
+		// No annotation declares a standalone LCOW
+		isSandboxOrStandalone = true
+	}
+
+	// Check if we have a network namespace and if so generate the resolv.conf
+	// so we can bindmount it into the container
+	var networkNamespace string
+	if settings.OCISpecification.Windows != nil &&
+		settings.OCISpecification.Windows.Network != nil &&
+		settings.OCISpecification.Windows.Network.NetworkNamespace != "" {
+		networkNamespace = strings.ToLower(settings.OCISpecification.Windows.Network.NetworkNamespace)
+	}
+	if networkNamespace != "" {
+		if isSandboxOrStandalone {
+			// We have a network namespace. Generate the resolv.conf and add the bind mount.
+			netopts := h.netNamespaces[networkNamespace]

```

Comment:
Should we check if the map actually has a value for this key before using it? Otherwise we'll panic when accessing netopts.Adapters below if the map doesn't have it.